src/v4/utils/utils.ts

```diff
@@ -108,13 +101,17 @@ export function isPlainObject(value) {
     return proto === Object.prototype || proto === null
 }
 
-export function convertToMap(dataStructure) {
+export function convertToMap(dataStructure: any): Map<any, any> {
     if (isES6Map(dataStructure) || isObservableMap(dataStructure)) {
         return dataStructure
     } else if (Array.isArray(dataStructure)) {
         return new Map(dataStructure)
     } else if (isPlainObject(dataStructure)) {
-        return new Map(Object.entries(dataStructure))
+        const map = new Map()
+        for (const key in dataStructure) {
```

Comment:
use `forOf` here?

Reply:
`forOf` is just a workaround for `Iterator`s - converting them to array is waste of resources and using `while` is PITA ... the function doesn't even accept/work with array/map/object.
Also I would prefer to keep this in sync with V5 (which intentionally doesn't have `forOf` util).

Reply:
Basically the idea is: If you use `for of` in V5, use native `.forEach` (array/map/set) or `forOf` for iterators in V4... `forOf/for of` for `object` doesn't make sense...